### DISCUSSION TITLE
Database Adjustment

### DISCUSSION THREAD

**Discussion by aljawaid** [creecros/MetaMagik#88] [Ideas]
I think this should be added to the plugin when creating a database table:
```
CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci
```
https://github.com/creecros/MetaMagik/blob/80af9abfe86dc7bb7a0d9c1ca0b98e6cb9f18446/Schema/Mysql.php#L65
https://github.com/creecros/MetaMagik/blob/80af9abfe86dc7bb7a0d9c1ca0b98e6cb9f18446/Schema/Mysql.php#L74

I noticed its not in the Postgres or SqLite files at all - not sure if its missing or meant to be like that.

**Comment by creecros** [comment]
Its MYSQL specific

**Comment by aljawaid** [comment]
> I think this should be added to the plugin when creating a database table:
> ```
> CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci
> ```
> https://github.com/creecros/MetaMagik/blob/80af9abfe86dc7bb7a0d9c1ca0b98e6cb9f18446/Schema/Mysql.php#L65
> 
> https://github.com/creecros/MetaMagik/blob/80af9abfe86dc7bb7a0d9c1ca0b98e6cb9f18446/Schema/Mysql.php#L74

`CHARSET` and `COLLATE` seems to need to be changed in the MySQL file though.

**Comment by creecros** [comment]
Im not going to pretend to know anything. I took my ques from Kanboard itself:
https://github.com/kanboard/kanboard/blob/1e304f4ad6e3c9f55f1c766a2d2e1614201e4a64/app/Schema/Mysql.php#L272

**Reply by creecros** [reply]
and i dont have any issues. are there issues i am unaware of?

**Reply by aljawaid** [reply]
look at line 133 in the same file.  It seems to be updated in later versions.

**Reply by aljawaid** [reply]
issues could be in updating and migrating.  When I manually migrated mid-last year, one of my issues were the mysql tables were all the wrong charset and collation... noticeable, my conflict was plugins and it confused the hell out of me on which was correct and which was wrong/corrupted.

No issues right now, but I noticed it and thought I would let you know.

**Reply by creecros** [reply]
i see it now, thanks for pointing it out, looks like it should be done.

**Reply by creecros** [reply]
https://github.com/creecros/MetaMagik/commit/d763613ff67473e0d3fa6a811ddd43a6c4b2f35d
